packages/relay/src/lib/clients/sdkClient.ts

```diff
@@ -643,60 +644,146 @@ export class SDKClient {
+      if (fileId && callData.length > this.fileAppendChunkSize) {
+        const fileAppendTx = await new FileAppendTransaction()
+          .setFileId(fileId)
+          .setContents(hexedCallData.substring(this.fileAppendChunkSize, hexedCallData.length))
+          .setChunkSize(this.fileAppendChunkSize)
+          .setMaxChunks(this.maxChunks);
+        const fileAppendTxResponse = await fileAppendTx.execute(client);
+
+        // get transaction fee and add expense to limiter
+        const appendFileRecord = await fileAppendTxResponse.getRecord(this.clientMain);
+        transactionFee = appendFileRecord.transactionFee;
+        this.hbarLimiter.addExpense(transactionFee.toTinybars().toNumber(), currentDateNow);
+
+        this.captureMetrics(
+          SDKClient.transactionMode,
+          fileAppendTx.constructor.name,
+          Status.Success,
+          await this.calculateFileAppendTxTotalTinybarsCost(fileAppendTx),
+          0,
+          callerName,
+          interactingEntity,
+        );
+      }
+
+      // Ensure that the calldata file is not empty
+      if (fileId) {
+        const fileSize = await (await new FileInfoQuery().setFileId(fileId).execute(client)).size;
 
-      if (callData.length > 0 && fileSize.isZero()) {
-        throw new SDKClientError({}, `${requestIdPrefix} Created file is empty. `);
+        if (callData.length > 0 && fileSize.isZero()) {
+          throw new SDKClientError({}, `${requestIdPrefix} Created file is empty. `);
+        }
+        this.logger.trace(`${requestIdPrefix} Created file with fileId: ${fileId} and file size ${fileSize}`);
       }
-      this.logger.trace(`${requestIdPrefix} Created file with fileId: ${fileId} and file size ${fileSize}`);
-    }
 
-    return fileId;
+      return fileId;
+    } catch (error: any) {
+      const sdkClientError = new SDKClientError(error, error.message);
+      let transactionFee: number | Hbar = 0;
+
+      // if valid network error utilize transaction id
+      if (sdkClientError.isValidNetworkError()) {
+        try {
+          const transactionCreateRecord = await new TransactionRecordQuery()
+            .setTransactionId(fileCreateTx.transactionId!)
+            .setNodeAccountIds(fileCreateTx.nodeAccountIds!)
+            .setValidateReceiptStatus(false)
+            .execute(this.clientMain);
+          transactionFee = transactionCreateRecord.transactionFee;
+          this.hbarLimiter.addExpense(transactionFee.toTinybars().toNumber(), currentDateNow);
+
+          this.captureMetrics(
+            SDKClient.transactionMode,
+            fileCreateTx.constructor.name,
+            sdkClientError.status,
+            transactionFee.toTinybars().toNumber(),
+            transactionCreateRecord?.contractFunctionResult?.gasUsed,
+            callerName,
+            interactingEntity,
+          );
+
+          if (fileAppendTx === null || fileAppendTx === undefined) {
```

Comment:
Below is what i'm suggesting.
Otherwise when you have an `fileAppendTx` you won't log the `fileCreateTx` details
```ts
            this.logger.info(
              `${requestIdPrefix} ${fileCreateTx.transactionId} ${callerName} ${fileCreateTx.constructor.name} status: ${sdkClientError.status} (${sdkClientError.status._code}), cost: ${transactionFee}`,
            );

        if (fileAppendTx) {
            const transactionAppendRecord = await new TransactionRecordQuery()
              .setTransactionId(fileAppendTx.transactionId!)
              .setNodeAccountIds(fileAppendTx.nodeAccountIds!)
              .setValidateReceiptStatus(false)
              .execute(this.clientMain);
            transactionFee = transactionAppendRecord.transactionFee;
            this.hbarLimiter.addExpense(transactionFee.toTinybars().toNumber(), currentDateNow);

            this.captureMetrics(
              SDKClient.transactionMode,
              fileCreateTx.constructor.name,
              sdkClientError.status,
              transactionFee.toTinybars().toNumber(),
              transactionCreateRecord?.contractFunctionResult?.gasUsed,
              callerName,
              interactingEntity,
            );

            this.logger.info(
              `${requestIdPrefix} ${fileAppendTx.transactionId} ${callerName} ${fileAppendTx.constructor.name} status: ${sdkClientError.status} (${sdkClientError.status._code}), cost: ${transactionFee}`,
            );
          }
```

Reply:
Done.